universal-login-react/src/ui/WalletSelector/Suggestions/Suggestions.tsx

```diff
@@ -1,35 +1,34 @@
 import React, {useState} from 'react';
-import {ensureNotNull, WalletSuggestionAction} from '@universal-login/commons';
+import {ensureNotNull, WalletSuggestionAction, Suggestions} from '@universal-login/commons';
 import {getSuggestion} from '../../../core/utils/getSuggestion';
 import {KeepTypingSuggestion} from './KeepTypingSuggestion';
 import {MissingParameter} from '../../../core/utils/errors';
 import {SingleSuggestion} from './SingleSuggestion';
 import {MultipleSuggestion} from './MultipleSuggestion';
 
 interface SuggestionsProps {
-  connections: string[];
-  creations: string[];
+  suggestions: Suggestions;
   source: string;
-  onCreateClick(ensName: string): Promise<void> | void;
-  onConnectClick(ensName: string): Promise<void> | void;
+  onCreateClick?(ensName: string): Promise<void> | void;
+  onConnectClick?(ensName: string): Promise<void> | void;
   actions: WalletSuggestionAction[];
 }
 
-export const Suggestions = ({connections, creations, onCreateClick, onConnectClick, actions, source}: SuggestionsProps) => {
+export const SuggestionsComponent = ({onCreateClick, onConnectClick, actions, source, suggestions}: SuggestionsProps) => {
   actions.includes(WalletSuggestionAction.connect) && ensureNotNull(onConnectClick, MissingParameter, 'onConnectClick');
   actions.includes(WalletSuggestionAction.create) && ensureNotNull(onCreateClick, MissingParameter, 'onCreateClick');
 
   const [selectedSuggestion, setSelectedSuggestion] = useState('');
   const handleConnectClick = (ensName: string) => {
     setSelectedSuggestion(ensName);
-    onConnectClick(ensName);
+    onConnectClick!(ensName);
   };
   const handleCreateClick = (ensName: string) => {
     setSelectedSuggestion(ensName);
-    onCreateClick(ensName);
+    onCreateClick!(ensName);
   };
 
-  const suggestion = getSuggestion(creations, connections, actions, source);
+  const suggestion = getSuggestion(suggestions, actions, source);
```

Comment:
Why not move this component up?
Now we first ask for suggestions, then we again ask for suggestions based on suggestions - sounds like we should merge it into one operation.

Next PR?